theano/scalar/basic.py

```diff
@@ -3099,7 +3109,8 @@ def init_name(self):
                 for i, r in enumerate(self.fgraph.variables):
                     if r not in io and len(r.clients) > 1:
                         r.name = 't%i' % i
-                rval = "Composite{%s}" % str(self.fgraph)
+                rval = "Composite{%s}" % ', '.join([pprint(output) for output
+                                                    in self.fgraph.outputs])
```

Comment:
By ugly output, I mean that we see the same computation multiple time. But
as my solution do not work, we can keep it like this. Just remember that
fact in case people have question about this.

On Wed, Jan 21, 2015 at 3:12 PM, Mohammad Pezeshki <notifications@github.com
> wrote:

> In theano/scalar/basic.py
> <https://github.com/Theano/Theano/pull/2407#discussion_r23330332>:
>
> > @@ -3099,7 +3109,8 @@ def init_name(self):
> >                  for i, r in enumerate(self.fgraph.variables):
> >                      if r not in io and len(r.clients) > 1:
> >                          r.name = 't%i' % i
> > -                rval = "Composite{%s}" % str(self.fgraph)
> > +                rval = "Composite{%s}" % ', '.join([pprint(output) for output
> > +                                                    in self.fgraph.outputs])
>
> @nouiz <https://github.com/nouiz> , I'm not sure that I understand your
> comment.
> But, when I write pprint(self.fgraph.outputs) I get this error:
>
> AttributeError: 'list' object has no attribute 'owner'
>
> By "ugly output", what exactly you mean?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Theano/Theano/pull/2407/files#r23330332>.
>